services/sinner/db/model/model.go

```diff
@@ -114,11 +114,11 @@ type Executed struct {
 	// BlockNumber is the block number in which the tx occurred.
 	BlockNumber uint64 `gorm:"column:block_number"`
 	// TxHash is the hash of the tx.
-	TxHash string `gorm:"column:tx_hash;primaryKey;index:idx_tx_hash_executed,priority:1,sort:desc"`
+	TxHash string `gorm:"column:tx_hash;index:idx_tx_hash_executed,priority:1,sort:desc"`
 	// TxIndex is the index of the tx in a block.
 	TxIndex uint `gorm:"column:tx_index"`
 	// MessageHash is the message hash.
-	MessageHash string `gorm:"column:message_hash"`
+	MessageHash string `gorm:"column:message_hash;primaryKey"`
```

Comment:
should be a foreign key constraint to message table

Reply:
Is this for data integrity if deleting from the db? currently the messagestatus table is updated concurrently w/origin/destination table insert while indexing. 

An idea could be to do a has many relationship, but the idea of the message status table was to be a lite as possible on query side so it could be pinged a lot for quick updates.

``` (has many)
OriginSentMessages []OriginSent `gorm:"foreignKey:MessageHash;references:MessageHash;constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
ExecutedMessages   []Executed   `gorm:"foreignKey:MessageHash;references:MessageHash;constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
```

Reply:
has many would be the idea yeah, this will also speed up joins, etc

Reply:
destination tx hash is many to may